### PR TITLE
[njpwworld] new extraction

### PR DESCRIPTION
**Pull request by aarubui**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] [Searched](https://github.com/ytdl-org/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Read [adding new extractor tutorial](https://github.com/ytdl-org/youtube-dl#adding-support-for-a-new-site)
- [x] Read [youtube-dl coding conventions](https://github.com/ytdl-org/youtube-dl#youtube-dl-coding-conventions) and adjusted the code to meet them
- [x] Covered the code with tests (note that PRs without tests will be REJECTED)
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Bug fix
- [ ] Improvement
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

Player code changed. Old extractor no longer works.

### CODE REVIEW

File: youtube_dl/extractor/njpwworld.py [hunk 1]
```diff
@@ -64,35 +71,27 @@ def _real_extract(self, url):
...
 
         self._sort_formats(formats)
 
-        post_content = get_element_by_class('post-content', webpage)
+        tag_block = get_element_by_class('tag-block', webpage)
         tags = re.findall(
-            r'<li[^>]+class="tag-[^"]+"><a[^>]*>([^<]+)</a></li>', post_content
-        ) if post_content else None
+            r'<a[^>]*>([^<]+)</a>', tag_block
+        )
```
Comment (remitamine) [inline youtube_dl/extractor/njpwworld.py:90]: keep similar checks for element class and `get_element_by_class` value.

File: youtube_dl/extractor/njpwworld.py [hunk 2]
```diff
@@ -64,35 +71,27 @@ def _real_extract(self, url):
...
+            })
 
         self._sort_formats(formats)
 
-        post_content = get_element_by_class('post-content', webpage)
+        tag_block = get_element_by_class('tag-block', webpage)
         tags = re.findall(
-            r'<li[^>]+class="tag-[^"]+"><a[^>]*>([^<]+)</a></li>', post_content
-        ) if post_content else None
+            r'<a[^>]*>([^<]+)</a>', tag_block
```
Comment (remitamine) [inline youtube_dl/extractor/njpwworld.py:89]: still the check for element class is missing.

Comment (aarubui) [reply youtube_dl/extractor/njpwworld.py]: I'm not understanding what you want. What do you mean by "element class"?

Comment (remitamine) [reply youtube_dl/extractor/njpwworld.py]: the `class` attribute of the `a` HTML element.

File: youtube_dl/extractor/njpwworld.py [hunk 3]
```diff
@@ -64,35 +71,28 @@ def _real_extract(self, url):
...
+                'ext': 'mp4',
+                'protocol': 'm3u8',
+                'quality': 2 if kind == 'high' else 1,
+            })
 
         self._sort_formats(formats)
 
-        post_content = get_element_by_class('post-content', webpage)
+        tag_block = get_element_by_class('tag-block', webpage)
+        tag_list = get_element_by_class('tag-list', tag_block) if tag_block else None
```
Comment (remitamine) [inline youtube_dl/extractor/njpwworld.py:88]: why is this extracted in two steps now, and not directly from `tag-block`?

File: youtube_dl/extractor/njpwworld.py [hunk 4]
```diff
@@ -64,35 +71,28 @@ def _real_extract(self, url):
...
         tags = re.findall(
-            r'<li[^>]+class="tag-[^"]+"><a[^>]*>([^<]+)</a></li>', post_content
-        ) if post_content else None
+            r'<a[^>]*\sclass="tag-[^>]*>([^<]+)</a>', tag_list
```
Comment (remitamine) [inline youtube_dl/extractor/njpwworld.py:90]:
```suggestion
            r'<a[^>]+class="tag-[^"]+"[^>]*>([^<]+)</a>', tag_list
```

File: youtube_dl/extractor/njpwworld.py (hunk 3 above)
Comment (aarubui) [reply youtube_dl/extractor/njpwworld.py:88]: Here's the HTML structure: Inside the `tag-block` div are the `tag-list` div (which has the `tag-` links) and a "More" link. This way was to exclude the "More" link in lieu of the "element class" check. Now we have a redundancy with both the `tag-list` and `tag-` filters.

Comment (remitamine) [reply youtube_dl/extractor/njpwworld.py:88]: after the change to check the `a` HTML element `class` attribute, this is no longer needed.

Comment (aarubui) [reply youtube_dl/extractor/njpwworld.py]: I already said it's a redundancy. It'd be clearer if you just said, "The `a` check is better than the `div` check. Go with that."

File: youtube_dl/extractor/njpwworld.py [hunk 5]
```diff
@@ -6,30 +6,37 @@
...
         'skip': 'Requires login',
-    }
+    }, {
+        'url': 'https://front.njpwworld.com/p/s_series_00563_16_bs',
+        'info_dict': {
+            'id': 's_series_00563_16_bs',
+            'ext': 'mp4',
+            'title': 'WORLD TAG LEAGUE 2020 & BEST OF THE SUPER Jr.27 2020年12月6日 福岡・福岡国際センター バックステージコメント（字幕あり）',
+            'tags': ["福岡・福岡国際センター", "バックステージコメント", "2020", "20年代"],
+        },
```
Comment (remitamine) [inline youtube_dl/extractor/njpwworld.py:38]: `skip_download` is needed for the test to pass similar to the first test.

**Pull request merged**